server/controllers/chartsController/helpers.test.js

```diff
@@ -739,4 +836,168 @@ describe('chartsController - helpers', () => {
       ).toEqual(2)
     })
   })
+
+  describe(helpers.buildFacetForFilters, () => {
+    it('returns an object to create lists for the requested filters', () => {
+      expect(
+        helpers.buildFacetForFilters({
+          isSexAtBirthFilterActive: true,
+          isInclusionCriteriaFilterActive: true,
+          userAccess,
+        })
+      ).toEqual({
+        socioDemographics: [
+          {
+            $match: {
+              assessment: SOCIODEMOGRAPHICS_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              socioDemographicsCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+        inclusionCriteria: [
+          {
+            $match: {
+              assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+      })
+    })
+
+    it('returns an object only for the requested filters', () => {
+      expect(
+        helpers.buildFacetForFilters({
+          isSexAtBirthFilterActive: true,
+          isInclusionCriteriaFilterActive: undefined,
+          userAccess,
+        })
+      ).toEqual({
+        socioDemographics: [
+          {
+            $match: {
+              assessment: SOCIODEMOGRAPHICS_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              socioDemographicsCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+      })
+    })
+  })
+
+  describe(helpers.intersectSubjectsFromFilters, () => {
+    it('adds subjects to the resultsFromFiltersMap', () => {
+      const requestedFilters = {
+        socioDemographics: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            socioDemographicsCollection:
+              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
+          },
+        ],
+        inclusionCriteria: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            inclusionCriteriaCollection:
+              '062419c48a8b6634e696b9dc0674fb87c34aea44751d1eba107feb0fc7325d71',
```

Comment:
I'm thinking more about how we could make this extensible to other filters. Would it make sense to create a standard object shape so we know what to expect across all filters and the `filter` key can determine what filter it is applied to?

```suggestion
            study: 'YA',
            subject: 'YA01508',
            collection: '062419c48a8b6634e696b9dc0674fb87c34aea44751d1eba107feb0fc7325d71',
            filter: 'inclusionCriteria',
```

Reply:
Yeah, but if this shape is the same across the filters, wouldn't they be replace during the intersection? I'm just thinking of the next step which is to query the individual collections

Reply:
oh I see. Yeah it would, unless we changed the value of the map like to an array of objects with this shape. Would that make sense given the next steps?

Reply:
Yeah, an array of objects could work. So the end result is something like
intersectedSubjectsMap = {
 "subjectA": {
    collections: [{collection: "filterACollection", filter:"filterA"},{collection: "filterBCollection", filter: "filterB" }],
    subject: "subjectA",
    study: "studyA"
    }
}

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/5cac1a4ad5d269fb70017befbad1aca505307a18

Reply:
do these tests need to be updated to reflect the new shape?

Reply:
they have been :) https://github.com/AMP-SCZ/dpdash/blob/5cac1a4ad5d269fb70017befbad1aca505307a18/server/controllers/chartsController/helpers.test.js

Reply:
but doesn't this test example still use the old structure?

Reply:
if this test is still passing I'm wonder what the value of it is since the data structure is no longer accurate (unless I'm missing something).

Reply:
I don't know, but I want to get this right.
These are the current data shapes.

The list of subjects for each filter:
```javascript
const filters  = {        
  socioDemographics: [
          {
            study: 'YA',
            subject: 'YA01508',
            collection:
              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
            filter: SOCIODEMOGRAPHICS_FORM,
          },
          {
            study: 'CA',
            subject: 'CA00063',
            collection: 'CA00063filter',
            filter: SOCIODEMOGRAPHICS_FORM,
          },
      ],
  }
```

When we intersect the lists of filters it looks like this 
```javascript
const intersectedSubject = {
     'YA01508': {
            subject: 'YA01508',
            collections: [
              {
                collection:
                  '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
                filter: SOCIODEMOGRAPHICS_FORM,
              },
              {
                collection:
                  '062419c48a8b6634e696b9dc0674fb87c34aea44751d1eba107feb0fc7325d71',
                filter: INCLUSION_EXCLUSION_CRITERIA_FORM,
              },
              {
                collection: 'thirdCriteriaFilterCollection',
                filter: 'thirdCriteria',
              },
            ],
          },
        ```
These are the shapes that are being tested for. What are your thoughts?

Reply:
I thought the keys `inclusionCriteriaCollection` and `socioDemographicsCollection` were renamed to `collection` and there was a new `filter` attribute? This setup data has the old shape.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/f239a2aa53315c7151b1c836cbabc16af5e2a3eb